tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Golden testing in Haskell with Hspec-Golden
+published: 2016-08-10
+tags: haskell, golden testing, snapshot testing, expect testing
+libraries: hspec-golden-1.0.0
+language: haskell
+author: Adrian Vera
+author-name: Adrian Vera
+twitter-profile: adrian_med
+github-profile: medAdrian
+description: A golden test is an IO action that writes its result to a file. To pass the test, this output file should be identical to the corresponding golden file, which contains the correct result for the test.
+---
+
+Golden tests are like unit tests, with the difference that the expected output is stored in a separate file. 
```

Comment:
@MedAdrian I think I've read that in a different [blog post about golden tests](https://ro-che.info/articles/2017-12-04-golden-tests). Perhaps, you got the idea from there?

> Golden tests are like unit tests, except the expected output is stored in a separate file.

I know the text isn't exactly the same, but we probably want to paraphrase to avoid copyright issues. Or, if you pick the sentence from that blog, we can probably reference it.

Reply:
I've paraphrased it (even tho i've really liked that phrase):

```
Golden testing is pretty similar to unit testing, except that the test output is stored in a separate file known as the "Golden File". 
```

How do i put it as a reference?
I'm thinking something like:

As seen on this blog that describes [Golden Testing](https://ro-che.info/articles/2017-12-04-golden-tests)

`Golden tests are like unit tests, except the expected output is stored in a separate file.`

Reply:
> Golden testing is pretty similar to unit testing, except that the test output is stored in a separate file known as the "Golden File". 

Sounds good.